### DISCUSSION TITLE
Update popper position on reference position change

### DISCUSSION THREAD

**Discussion by tounsoo** [floating-ui/floating-ui#1263] [Q&A]
I can't seem to find a way to update the popper position when the reference position changes. 

For example, if I had a popper opened inside a card that can be moved down when the card above gets longer, the popper position is not updating unless I scroll or change the window size.

HELP!?! @.@

> using react-popper

**Comment by FezVrasta** [comment]
You can call the `updatePopper` function to update the popper position.

**Reply by tounsoo** [reply]
But I don't know how to detect reference position change.

**Reply by FezVrasta** [reply]
That's responsibility of your application code, Popper can't know it for you. I suppose you have some drag and drop system that allows you to move the reference element? If so, you should call `updatePopper` any time the drag-n-drop system detects a drag event.

**Reply by tounsoo** [reply]
I work with creating desgin-system and I'm using react-popper for a Popover component. I can't rely on external people to call `updatePopper`. 😢

**Reply by FezVrasta** [reply]
You could add a MutationObserver and a ResizeObserver so that anytime the inline `style` of the element or the size of the element changes you trigger an `updatePopper` call.

**Reply by tounsoo** [reply]
Hmmm I might be wrong, but if the MutationObserver only knows how to check `style`, it won't work since the parent element would be moving -- `style` attribute of the reference is not changing. 

I'm currently using ResizeObserver to handle width change of the referenceElement but that one is not triggered when the position of the reference element changes.

I believe I need the offset values to detect the position change.
